### PR TITLE
Working Build Menu and more

### PR DESCRIPTION
**Pull request by JakeGilla-DL**
- Camera stops processing input when mouse leaves the viewport  
- Cleaned up some early prototype files
- Light reoganization of a few files

### CODE REVIEW

File: scripts/buildings/building.gd [hunk 1]
```diff
@@ -28,18 +33,21 @@ func _ready() -> void:
 	_generate_states()
 	if building_data == null:
 		state_machine.change_state("Empty")
-	elif is_prebuilt:
```
Comment (SlowFaller) [inline scripts/buildings/building.gd:31]: Is there no edge case for just plopping down a building outside of the queue in a pre-built state?

File: scripts/buildings/building.gd [hunk 2]
```diff
@@ -67,32 +75,62 @@ func show_waiting_dots(p_should_show: bool) -> void:
...
 func _on_build_button_pressed() -> void:
+	state_machine.change_state("MenuState")
 	build_button_pressed.emit(self)
+
+func _on_building_selected(p_building_data: RBuilding) -> void:
+	building_data = p_building_data
+	show_building_texture(true)
+	show_scaffolding(true)
+
+func _on_menu_cancelded() -> void:
```
Comment (SlowFaller) [inline scripts/buildings/building.gd]: Should be either cancelled or canceled, there is an extra D in there, and maybe a missing L

File: scripts/menus/build_menu.gd [hunk 3]
```diff
@@ -1,34 +1,82 @@
...
 
+@export_group("Tree References")
+@export var material_warehouse: MaterialWarehouse
+
 var active_building: Building
+var available_buildings: Array[RBuilding]
 
-signal build_pressed(p_selected_building: RBuilding)
 signal building_selected(p_selected_building: RBuilding)
+signal menu_caneceled()
```
Comment (SlowFaller) [inline scripts/menus/build_menu.gd]: Same here, there is an extra E, and depending on how you write it, a missing L. Should be canceled or cancelled.

**Comment by SlowFaller** [comment]
Gave it a once over, mostly for grammatical stuff and looking for adherence to our set standard. I still want to run the game, and do a deeper analysis of the code logic.

File: scripts/states/building_states/menu_state.gd [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+	parent_building.show_building_texture(false)
+	parent_building.show_building_flag(false)
+	parent_building.show_progress_bar(false)
+	parent_building.show_waiting_dots(false)
+	parent_building.show_scaffolding(false)
+	parent_building.show_build_button(false)
+	parent_building.fade_building(true)
+	parent_building.set_building_name()
+
+# Called every frame when this state is active.
```
Comment (SlowFaller) [inline scripts/states/building_states/menu_state.gd]: Remove unneeded functions from the states.

File: scripts/buildings/building.gd (hunk 1 above)
Comment (JakeGilla-DL) [reply scripts/buildings/building.gd:31]: Scaffolding probably, we should make it a thing, but will do a story for this

**Review by SlowFaller — CHANGES_REQUESTED** [review]
Some cleanup of empty functions, otherwise looks good.

**Comment by SlowFaller** [comment]
PR approved, changes requested were made.

**Review by SlowFaller — APPROVED** [review]
Approved.

**Pull request merged**